### DISCUSSION TITLE
Is there a default parameter to allow a sidebar or table of contents?

### DISCUSSION THREAD

**Discussion by CodyBontecou** [slidevjs/slidev#439] [Q&A] [answered]
I'd like to have sidebar navigation allowing the user to navigate to whichever slide he/she wants to see. Is there a config parameter that I missed allowing this?

**Comment by yjl9903** [comment]
You can press `o` to the slides overview page for quick navigating.

See https://sli.dev/guide/navigation.html#slides-overview

**Comment by antfu** [comment, marked as answer]
You might need to implement one yourself atm. You can use the [Global Context](https://sli.dev/custom/vue-context.html) to get the slides info in Vue and render them as you want

**Reply by CodyBontecou** [reply]
Ah okay, it looks like that's what it'll take. Thank you for the response.

**Reply by tobiasBora** [reply]
It would be great to have an official method to deal with table of contents, so that we can design styles accordingly. For instance, in LaTeX/Beamer you can display the table of content in the header of each slide to visualize the progression in the sections as [shown here](https://deic.uab.cat/~iblanes/beamer_gallery/large/Darmstadt-default-default-02.png).

**Reply by tonai** [reply]
Hello @tobiasBora maybe you can be interested in this plugin made for slidev: https://github.com/Smile-SA/slidev-component-progress

**Comment by tonai** [comment]
I'm also interested in having some Table Of Content but in the `$slidev` Global Context I don't see any information about the slide content itself.
Is there a way to extract the slide title or slide content ?

**Comment by tonai** [comment]
@antfu I can create a PR for a built-in TOC component if you want, but I am wondering one thing, is it possible to access the `$slidev` inside the `<script setup>` ?
